src/Components/ServiceScene/Breakdowns/SelectLabelActionScene.tsx

```diff
@@ -1,24 +1,177 @@
-import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
-import { ServiceScene } from '../ServiceScene';
+import {
+  AdHocFiltersVariable,
+  SceneComponentProps,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  VizPanel,
+} from '@grafana/scenes';
+import { getLogsPanelFrame, ServiceScene } from '../ServiceScene';
 import { navigateToValueBreakdown } from '../../../services/navigate';
 import { ValueSlugs } from '../../../services/routing';
 import { Button } from '@grafana/ui';
 import React from 'react';
+import { addToFilters } from './AddToFiltersButton';
+import { FilterButton } from '../../FilterButton';
+import {
+  EMPTY_VARIABLE_VALUE,
+  getFieldsVariable,
+  getLabelsVariable,
+  getLevelsVariable,
+  LEVEL_VARIABLE_VALUE,
+  SERVICE_NAME,
+} from '../../../services/variables';
+import { AdHocVariableFilter, Field, Labels, LoadingState } from '@grafana/data';
+import { FilterOp } from '../../../services/filters';
 
-interface SelectLabelActionSceneState extends SceneObjectState {
+interface SelectFieldActionSceneState extends SceneObjectState {
   labelName: string;
+  fieldType: ValueSlugs;
+  hideValueDrilldown?: boolean;
+  showFilterField?: boolean;
 }
-export class SelectLabelActionScene extends SceneObjectBase<SelectLabelActionSceneState> {
-  public onClick = () => {
-    const serviceScene = sceneGraph.getAncestor(this, ServiceScene);
-    navigateToValueBreakdown(ValueSlugs.label, this.state.labelName, serviceScene);
-  };
+
+export class SelectLabelActionScene extends SceneObjectBase<SelectFieldActionSceneState> {
+  constructor(state: SelectFieldActionSceneState) {
+    super(state);
+    this.addActivationHandler(this.onActivate.bind(this));
+  }
 
   public static Component = ({ model }: SceneComponentProps<SelectLabelActionScene>) => {
+    const { hideValueDrilldown, labelName, showFilterField } = model.useState();
+    const variable = model.getVariable();
+    const variableName = variable.useState().name;
+    const existingFilter = model.getExistingFilter(variable);
+
     return (
-      <Button variant="secondary" size="sm" onClick={model.onClick} aria-label={`Select ${model.useState().labelName}`}>
-        Select
-      </Button>
+      <>
+        {variable && showFilterField === true && (
```

Comment:
If variable could not be defined, then `const variableName = variable.useState().name;` (line 43) should throw. So either one or the other.